rust/src/milla/simulate.rs

```diff
@@ -3,158 +3,389 @@ use crate::milla::model::*;
-/// For a tile losing air equally in all directions, flow will be zero.
-/// For a tile losing air towards positive only, flow will be the air pressure moved.
-pub(crate) fn calculate_flow(
-    maybe_negative_tile: Option<&Tile>,
-    tile: &Tile,
-    maybe_positive_tile: Option<&Tile>,
-    is_x: bool,
-) -> f32 {
-    let mut flow = 0.0;
-    let my_pressure = tile.pressure();
-    if let Some(negative_tile) = maybe_negative_tile {
-        // Make sure air can pass this way.
-        let mut permeable = true;
-        if is_x {
-            permeable = permeable
-                && !negative_tile
+            let (my_tile, their_tile) = next.get_pair_mut(my_index, their_index);
+            if *dx > 0
+                && (my_tile
                     .airtight_directions
-                    .contains(AirtightDirections::EAST);
-            permeable = permeable && !tile.airtight_directions.contains(AirtightDirections::WEST);
-        } else {
-            permeable = permeable
-                && !negative_tile
+                    .contains(AirtightDirections::EAST)
+                    || their_tile
+                        .airtight_directions
+                        .contains(AirtightDirections::WEST))
+            {
+                // Something's blocking airflow.
+                my_tile.wall[axis] = true;
+                continue;
+            } else if *dy > 0
+                && (my_tile
                     .airtight_directions
-                    .contains(AirtightDirections::NORTH);
-            permeable = permeable && !tile.airtight_directions.contains(AirtightDirections::SOUTH);
+                    .contains(AirtightDirections::NORTH)
+                    || their_tile
+                        .airtight_directions
+                        .contains(AirtightDirections::SOUTH))
+            {
+                // Something's blocking airflow.
+                my_tile.wall[axis] = true;
+                continue;
+            }
+            my_tile.wall[axis] = false;
         }
-        if permeable {
-            let negative_pressure = negative_tile.pressure();
-            if negative_pressure < my_pressure {
-                // Causes flow in the negative direction.
-                flow += negative_pressure - my_pressure;
+    }
+}
+
+/// Calculate the new wind at each boundary.
+pub(crate) fn update_wind(prev: &ZLevel, next: &mut ZLevel) {
+    for my_index in 0..MAP_SIZE * MAP_SIZE {
+        let x = (my_index / MAP_SIZE) as i32;
+        let y = (my_index % MAP_SIZE) as i32;
+        let my_tile = prev.get_tile(my_index);
+
+        for (axis, (dx, dy)) in AXES.iter().enumerate() {
+            let maybe_neighbor_index = ZLevel::maybe_get_index(x + dx, y + dy);
+            let neighbor_index;
+            match maybe_neighbor_index {
+                Some(index) => neighbor_index = index,
+                None => continue,
+            }
```

Comment:
Saves you a variable initialization without losing too much clarity
```suggestion
            let neighbor_index = match ZLevel::maybe_get_index(x + dx, y + dy) {
                Some(index) => index,
                None => continue,
            }
```